wpimath/src/main/java/edu/wpi/first/math/filter/SlewRateLimiter.java

```diff
@@ -14,53 +14,115 @@
-   * Filters the input to limit its slew rate.
-   *
-   * @param input The input value whose slew rate is to be limited.
-   * @return The filtered value, which will not change faster than the slew rate.
-   */
-  public double calculate(double input) {
-    double currentTime = WPIUtilJNI.now() * 1e-6;
-    double elapsedTime = currentTime - m_prevTime;
-    m_prevVal +=
-        MathUtil.clamp(input - m_prevVal, -m_rateLimit * elapsedTime, m_rateLimit * elapsedTime);
-    m_prevTime = currentTime;
-    return m_prevVal;
-  }
+    /**
+     * Creates a new SlewRateLimiter with the given forward rate and backward rate.
+     *
+     * @param forward_Rate
+     * @param backwardRateLimit
+     */
+    public SlewRateLimiter(double forwardRateLimit, double backwardRateLimit) {
+        m_forwardRateLimit = forwardRateLimit;
+        m_backwardRateLimit = backwardRateLimit;
+        m_prevTime = WPIUtilJNI.now() * 1e-6;
+    }
 
-  /**
-   * Resets the slew rate limiter to the specified value; ignores the rate limit when doing so.
-   *
-   * @param value The value to reset to.
-   */
-  public void reset(double value) {
-    m_prevVal = value;
-    m_prevTime = WPIUtilJNI.now() * 1e-6;
-  }
 }
+
+    /**
+     * Filters the input to limit its slew rate.
+     *
+     * @param input The input value whose slew rate is to be limited.
+     * @return The filtered value, which will not change faster than the slew rate.
+     */
+    public double calculate(double input) {
+        double currentTime = WPIUtilJNI.now() * 1e-6;
+        double elapsedTime = currentTime - m_prevTime;
+        m_prevVal +=
+                MathUtil.clamp(input - m_prevVal, -m_rateLimit * elapsedTime, m_rateLimit * elapsedTime);
+        m_prevTime = currentTime;
+        return m_prevVal;
+    }
+
+    public double calculate(double input) {
+        double currentTime = WPIUtilJNI.now() * 1e-6;
+        double elapsedTime = currentTime - m_prevTime;
+        m_prevVal +=
+                MathUtil.clamp(input - m_prevVal, -m_rateLimit * elapsedTime, m_rateLimit * elapsedTime);
+        m_prevTime = currentTime;
+        return m_prevVal;
+    }
+    /**
+     * Filters the input to limit the slew rate for backward and forward.
+     *
+     * @param input The input value whose slew rate is to be limited.
+     */
+    public double calculate(double input) {
+        double currentTime = WPIUtilJNI.now() * 1e-6;
+        double elapsedTime = currentTime - m_prevTime;
+        m_prevVal +=
+                MathUtil.clamp(input - m_prevVal, -m_backwardRateLimit * elapsedTime, m_forwardRateLimit * elapsedTime);
+        m_prevTime = currentTime;
+        return m_prevVal;
+    }
+
+    /**
+     * Resets the slew rate limiter to the specified value; ignores the rate limit when doing so.
+     *
+     * @param value The value to reset to.
+     */
+    public void reset(double value) {
+        m_prevVal = value;
+        m_prevTime = WPIUtilJNI.now() * 1e-6;
+    }
+    /**
```

Comment:
Small thing, but formatting might complain about no empty lines between methods.

Reply:
The entire file needs to be reformatted (`.\gradlew spotlessApply`, I think) so it's indented with two spaces and everything.